src/sentry/search/events/builder/metrics.py

```diff
@@ -1510,6 +1500,60 @@ def run_query(self, referrer: str, use_cache: bool = False) -> Any:
             "meta": [{"name": key, "type": value} for key, value in meta_dict.items()],
         }
 
+    def _metric_layer_result(
+        self, metrics_data_list: Sequence[dict[str, Any]], use_first_group_only: bool
+    ) -> Any:
+        """The metric_layer returns results in a non-standard format, this function changes it back to the expected
+        one"""
+        metric_layer_result: Any = {
+            "data": [],
+            "meta": [],
+        }
+
+        seen_metrics_metas = {}
+        time_data_map = defaultdict(dict)
+
+        for metrics_data in metrics_data_list:
+            for meta in metrics_data["meta"]:
+                if meta["name"] not in seen_metrics_metas:
+                    seen_metrics_metas[meta["name"]] = True
+                    metric_layer_result["meta"].append(meta)
+            for meta in metric_layer_result["meta"]:
+                if meta["name"] == "bucketed_time":
+                    meta["name"] = "time"
+
+            # Top timeseries uses all groups, but regular timeseries only need the first group.
+            _groups = metrics_data.get("groups", [{}])
+            groups = _groups[:1] if use_first_group_only else _groups
```

Comment:
The only difference converting metric layer result between TopTimeseries and regular Timeseries

Reply:
Good refactor 👍🏻